### ISSUE TITLE
PHP 7.2 does not satisfy the requirements

### ISSUE THREAD

**Issue by edwinsiebel** [nunomaduro/phpinsights#396]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| Library version  | v1.14.0

Running docker with php 7.2 (which should be [supported](https://phpinsights.com/get-started.html#within-laravel)), on an empty composer install: `composer require nunomaduro/phpinsights --dev`.

However, composer complains about that my php version does not satisfy my requirement.

```
Problem 1
    - phploc/phploc 6.0.2 requires php ^7.3 -> your PHP version (7.2.29) does not satisfy that requirement.
    - phploc/phploc 6.0.1 requires php ^7.3 -> your PHP version (7.2.29) does not satisfy that requirement.
    - phploc/phploc 6.0.0 requires php ^7.3 -> your PHP version (7.2.29) does not satisfy that requirement.
    - Installation request for nunomaduro/phpinsights ^1.14 -> satisfiable by nunomaduro/phpinsights[v1.14.0].
    - nunomaduro/phpinsights v1.14.0 requires phploc/phploc ^5.0|^6.0 -> satisfiable by phploc/phploc[5.0.0, 6.0.0, 6.0.1, 6.0.2].
    - Conclusion: remove symfony/console v5.0.7
    - phploc/phploc 5.0.0 requires symfony/console ^4.0 -> satisfiable by symfony/console[v4.0.0, v4.0.1, v4.0.10, v4.0.11, v4.0.12, v4.0.13, v4.0.14, v4.0.15, v4.0.2, v4.0.3, v4.0.4, v4.0.5, v4.0.6, v4.0.7, v4.0.8, v4.0.9, v4.1.0, v4.1.1, v4.1.10, v4.1.11, v4.1.12, v4.1.2, v4.1.3, v4.1.4, v4.1.5, v4.1.6, v4.1.7, v4.1.8, v4.1.9, v4.2.0, v4.2.1, v4.2.10, v4.2.11, v4.2.12, v4.2.2, v4.2.3, v4.2.4, v4.2.5, v4.2.6, v4.2.7, v4.2.8, v4.2.9, v4.3.0, v4.3.1, v4.3.10, v4.3.11, v4.3.2, v4.3.3, v4.3.4, v4.3.5, v4.3.6, v4.3.7, v4.3.8, v4.3.9, v4.4.0, v4.4.1, v4.4.2, v4.4.3, v4.4.4, v4.4.5, v4.4.6, v4.4.7].
    - Can only install one of: symfony/console[v5.0.7, v4.1.10].
    - Can only install one of: symfony/console[v5.0.7, v4.1.11].
    - Can only install one of: symfony/console[v5.0.7, v4.1.12].
    - Can only install one of: symfony/console[v5.0.7, v4.1.6].
    - Can only install one of: symfony/console[v5.0.7, v4.1.7].
    - Can only install one of: symfony/console[v5.0.7, v4.1.8].
    - Can only install one of: symfony/console[v5.0.7, v4.1.9].
    - Can only install one of: symfony/console[v5.0.7, v4.2.0].
    - Can only install one of: symfony/console[v5.0.7, v4.2.1].
    - Can only install one of: symfony/console[v5.0.7, v4.2.10].
    - Can only install one of: symfony/console[v5.0.7, v4.2.11].
    - Can only install one of: symfony/console[v5.0.7, v4.2.12].
    - Can only install one of: symfony/console[v5.0.7, v4.2.2].
    - Can only install one of: symfony/console[v5.0.7, v4.2.3].
    - Can only install one of: symfony/console[v5.0.7, v4.2.4].
    - Can only install one of: symfony/console[v5.0.7, v4.2.5].
    - Can only install one of: symfony/console[v5.0.7, v4.2.6].
    - Can only install one of: symfony/console[v5.0.7, v4.2.7].
    - Can only install one of: symfony/console[v5.0.7, v4.2.8].
    - Can only install one of: symfony/console[v5.0.7, v4.2.9].
    - Can only install one of: symfony/console[v5.0.7, v4.3.0].
    - Can only install one of: symfony/console[v5.0.7, v4.3.1].
    - Can only install one of: symfony/console[v5.0.7, v4.3.10].
    - Can only install one of: symfony/console[v5.0.7, v4.3.11].
    - Can only install one of: symfony/console[v5.0.7, v4.3.2].
    - Can only install one of: symfony/console[v5.0.7, v4.3.3].
    - Can only install one of: symfony/console[v5.0.7, v4.3.4].
    - Can only install one of: symfony/console[v5.0.7, v4.3.5].
    - Can only install one of: symfony/console[v5.0.7, v4.3.6].
    - Can only install one of: symfony/console[v5.0.7, v4.3.7].
    - Can only install one of: symfony/console[v5.0.7, v4.3.8].
    - Can only install one of: symfony/console[v5.0.7, v4.3.9].
    - Can only install one of: symfony/console[v5.0.7, v4.4.0].
    - Can only install one of: symfony/console[v5.0.7, v4.4.1].
    - Can only install one of: symfony/console[v5.0.7, v4.4.2].
    - Can only install one of: symfony/console[v5.0.7, v4.4.3].
    - Can only install one of: symfony/console[v5.0.7, v4.4.4].
    - Can only install one of: symfony/console[v5.0.7, v4.4.5].
    - Can only install one of: symfony/console[v5.0.7, v4.4.6].
    - Can only install one of: symfony/console[v5.0.7, v4.4.7].
    - Can only install one of: symfony/console[v5.0.7, v4.0.0].
    - Can only install one of: symfony/console[v5.0.7, v4.0.1].
    - Can only install one of: symfony/console[v5.0.7, v4.0.10].
    - Can only install one of: symfony/console[v5.0.7, v4.0.11].
    - Can only install one of: symfony/console[v5.0.7, v4.0.12].
    - Can only install one of: symfony/console[v5.0.7, v4.0.13].
    - Can only install one of: symfony/console[v5.0.7, v4.0.14].
    - Can only install one of: symfony/console[v5.0.7, v4.0.15].
    - Can only install one of: symfony/console[v5.0.7, v4.0.2].
    - Can only install one of: symfony/console[v5.0.7, v4.0.3].
    - Can only install one of: symfony/console[v5.0.7, v4.0.4].
    - Can only install one of: symfony/console[v5.0.7, v4.0.5].
    - Can only install one of: symfony/console[v5.0.7, v4.0.6].
    - Can only install one of: symfony/console[v5.0.7, v4.0.7].
    - Can only install one of: symfony/console[v5.0.7, v4.0.8].
    - Can only install one of: symfony/console[v5.0.7, v4.0.9].
    - Can only install one of: symfony/console[v5.0.7, v4.1.0].
    - Can only install one of: symfony/console[v5.0.7, v4.1.1].
    - Can only install one of: symfony/console[v5.0.7, v4.1.2].
    - Can only install one of: symfony/console[v5.0.7, v4.1.3].
    - Can only install one of: symfony/console[v5.0.7, v4.1.4].
    - Can only install one of: symfony/console[v5.0.7, v4.1.5].
    - Installation request for symfony/console (installed at v5.0.7) -> satisfiable by symfony/console[v5.0.7].
```

**Comment by olivernybroe** [comment]
When I read this stack that is not what I get from it. 

What I am getting is that it is trying to install `phploc/phploc`, it can choose between v6 or v5, but v6 requires php7.3, so it now tried with v5.

phploc v5 requires symfony/console v4, so symfony/console v5 cannot be installed.

What I am getting from this message is that in your composer you have required symfony/console v5, which simply is not possible. Could you share your composer file? :)

**Comment by 50bhan** [comment]
@edwinsiebel https://github.com/nunomaduro/phpinsights/issues/394#issuecomment-617018805

**Comment by edwinsiebel** [comment]
Hi @50bhan, yes, that works perfect. Got it installed quick and easy.

Thanks for the update.

**Comment by fabzster** [comment]
I am getting this and I have php 8.3.9:

`Problem 1
    - nunomaduro/phpinsights[v1.0.0, ..., v1.14.1] require php ^7.2 -> your php version (8.3.9) does not satisfy that requirement.
    - nunomaduro/phpinsights[v2.0.0, ..., v2.0.1] require psr/container ^1.0 -> found psr/container[1.0.0, 1.1.0, 1.1.1, 1.1.2] but the package is fixed to 2.0.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - nunomaduro/phpinsights[v2.1.0, ..., v2.3.0] require psr/simple-cache ^1.0 -> found psr/simple-cache[1.0.0, 1.0.1] but the package is fixed to 3.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - nunomaduro/phpinsights[v2.4.0, ..., v2.5.0] require symfony/console ^4.2|^5.0|^6.0 -> found symfony/console[v4.2.0, ..., v4.4.49, v5.0.0, ..., v5.4.41, v6.0.0, ..., v6.4.9] but the package is fixed to v7.1.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - nunomaduro/phpinsights[v2.6.0, ..., v2.7.0] require sebastian/diff ^4.0 -> found sebastian/diff[4.0.0, ..., 4.0.6] but the package is fixed to 6.0.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - nunomaduro/phpinsights[v2.8.0, ..., v2.10.0] require sebastian/diff ^4.0|^5.0 -> found sebastian/diff[4.0.0, ..., 4.0.6, 5.0.0, ..., 5.1.1] but the package is fixed to 6.0.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - nunomaduro/phpinsights v2.11.0 requires sebastian/diff ^4.0|^5.0.3 -> found sebastian/diff[4.0.0, ..., 4.0.6, 5.0.3, 5.1.0, 5.1.1] but the package is fixed to 6.0.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires nunomaduro/phpinsights * -> satisfiable by nunomaduro/phpinsights[v1.0.0, ..., v1.14.1, v2.0.0, ..., v2.11.0].`
